core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java

```diff
@@ -128,10 +122,25 @@ static void initializeSettings(final Settings.Builder output, final Settings inp
     }
 
     /**
-     * Finish preparing settings by replacing forced settings, prompts, and any defaults that need to be added.
-     * The provided terminal is used to prompt for settings needing to be replaced.
+     * Checks all settings values to make sure they do not have the old prompt settings. These were deprecated in 6.0.0.
      */
-    private static void finalizeSettings(Settings.Builder output, Terminal terminal) {
+    private static void checkSettingsForTerminalDeprecation(final Settings.Builder output) throws SettingsException {
+        for (String setting : output.keys()) {
```

Comment:
Let's add an assertion here the major property of the current version is not 8; then, when we bump the version to 8.0.0 on master we know that we can remove this code.

Reply:
dude, nice. Thats smart. Will do.